lib/src/boilerplate_suggestors/annotations_remover.dart

```diff
@@ -16,12 +16,75 @@ import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:codemod/codemod.dart';
 
-/// Suggestor that looks for @Props, @State, and @Component2 and removes them.
+import 'boilerplate_utilities.dart';
+
+/// Suggestor that looks for `@Props()`, `@State()`, `@Component2()` and `@Factory()` annotations, and removes them
+/// as long as they do not contain any arguments.
+///
+/// > NOTE: `@PropsMixin()` and `@StateMixin()` annotations are removed via [PropsMixinMigrator].
 class AnnotationsRemover extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
   @override
   visitAnnotatedNode(AnnotatedNode node) {
     super.visitAnnotatedNode(node);
+    if (node.metadata.isEmpty) return;
+
+    final annotationToRemove =
+        node.metadata.singleWhere(_annotationIsRelevant, orElse: () => null);
+
+    // --- Short Circuit Conditions --- //
+    if (annotationToRemove == null) return;
+    if (annotationToRemove.arguments.arguments.isNotEmpty) return;
+    if (!_propsClassWasConvertedToNewBoilerplate(node)) return;
+
+    // --- Migrate --- //
+    yieldPatch(annotationToRemove.offset,
+        node.firstTokenAfterCommentAndMetadata.offset, '');
```

Comment:
This could potentially remove other code (e.g., other annotations) between the relevant annotation and the class declarations. Dartfmt will handle adjusting newlines/whitespace as needed.
```suggestion
        annotationToRemove.end, '');
```